### PR TITLE
🔒️♻️(edx) prevent deletion of entries from protected edx tables and refactor mixin

### PR DESCRIPTION
**Pull request by wilbrdt**
## Purpose

- Certain edx tables should not allow deletion. Rather than depending solely on the ON DELETE RESTRICT constraint in the database, we handle this with SQLAlchemy within Mork.
- The `AuthUserMixin` class adds unnecessary complexity and won't be reused accross other models.

## Proposal

- [x] To simplify the code, let's remove the mixin and refactor its methods into standalone CRUD functions.
- [x] A check is added to ensure that, if the user being deleted has associated entries in these protected tables, an error is raised, preventing the deletion.

Note:
- [x] Adding ability to remove entries from the `student_courseenrollmentallowed` as it lacks foreign key relationships with other tables.

### CODE REVIEW

File: src/app/mork/edx/crud.py
Comment (quitterie-lcs) [inline src/app/mork/edx/crud.py:1]: Why do you think it is a better solution to have standalone functions rather than a mixin ?

Comment (wilbrdt) [reply src/app/mork/edx/crud.py:1]: I think it adds some unnecessary complexity and that the methods defined in the `AuthUserMixin` won't be reused by any other models than `AuthUser`

**Review by lunika — APPROVED** [review]

File: src/app/mork/edx/crud.py
Comment (quitterie-lcs) [reply src/app/mork/edx/crud.py:1]: I would have proposed defining crud methods as 'AuthUser' class method for consistency and business logic encapsulation. However, in the current context, `AuthUser` SQL Alchemy model is already difficult to read with columns and relations definitions. To keep it simple, I agree with the design you proposed.

**Review by quitterie-lcs — APPROVED** [review]

**Pull request merged**